// react-native-calendars/src/dayAgenda/reservation-list/style.js
import {StyleSheet} from 'react-native';
import * as defaultStyle from '../../style';
export default function styleConstructor(theme = {}) {
  const appStyle = {...defaultStyle, ...theme};
  return StyleSheet.create({
    container: {
      flexDirection: 'row',
    },
    innerContainer: {
      flex: 1,
    },
    dayNum: {
      fontSize: 22,
      fontWeight: '200',
      fontFamily: appStyle.textDayFontFamily,
      color: appStyle.agendaDayNumColor,
    },
    hourNum: {
      // fontSize: 28,
      fontSize: 22,
      fontWeight: '200',
      fontFamily: appStyle.textDayFontFamily,
      color: appStyle.agendaDayNumColor,
    },
    hourText: {
      fontSize: 14,
      fontWeight: appStyle.textDayFontWeight,
      fontFamily: appStyle.textDayFontFamily,
      color: appStyle.agendaDayTextColor,
      backgroundColor: 'rgba(0,0,0,0)',
      marginTop: -5,
    },
    dayText: {
      fontSize: 14,
      fontWeight: appStyle.textDayFontWeight,
      fontFamily: appStyle.textDayFontFamily,
      color: appStyle.agendaDayTextColor,
      backgroundColor: 'rgba(0,0,0,0)',
      marginTop: -5,
    },
    day: {
      width: 63,
      alignItems: 'center',
      justifyContent: 'flex-start',
      marginTop: 32,
    },
    hour: {
      width: 63,
      alignItems: 'center',
      justifyContent: 'flex-start',
      marginTop: 32,
    },
    today: {
      color: appStyle.agendaTodayColor,
    },
    indicator: {
      marginTop: 80,
    },
    // @ts-expect-error
    ...(theme['stylesheet.agenda.list'] || {}),
  });
}
